k9mail-library/src/main/java/com/fsck/k9/mail/oauth/OAuth2TokenProvider.java

```diff
@@ -0,0 +1,61 @@
+package com.fsck.k9.mail.oauth;
+
+import android.app.Activity;
+
+import com.fsck.k9.mail.AuthenticationFailedException;
+
+import java.util.List;
+
+public interface OAuth2TokenProvider {
```

Comment:
That's the expectation, yes. 

An (imperfect) implementation that supports GMail only is here: https://github.com/philipwhiuk/k-9/commit/7bd0f52c6d24ba1696620c4109c30d1f06aa770f#diff-4e75c10b7bac153fec6888b448a738ea